cmd/root.go

```diff
@@ -55,6 +56,36 @@ func RunCommand() *cobra.Command {
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
 			output.SetOut(cmd.OutOrStdout())
 			output.SetErr(cmd.ErrOrStderr())
+
+			// Load config from the file (or default if it doesn't exist)
+			configHome, err := getConfigHome(opts)
+			if err != nil {
+				output.Errorf("Failed to read base config directory")
+				output.Infof("Use the --config flag or set the $%s environment variable to provide a default", constants.KitopsHomeEnvVar)
+				output.Debugf("Error: %s", err)
+				return errors.New("exit")
+			}
+
+			configPath := filepath.Join(configHome, "config.json")
+			cfg, err := config.LoadConfig(configPath)
+			if err != nil {
+				if !os.IsNotExist(err) { // If the config file exists but there's an error, report it
+					return output.Fatalf("Failed to load config: %s", err)
+				}
+				cfg = config.DefaultConfig() // Load default config if file doesn't exist
+			}
+
+			// Override the flags with config values if present
+			if cfg.LogLevel != "" {
+				opts.loglevel = cfg.LogLevel
+			}
+			if cfg.Progress != "" {
+				opts.progressBars = cfg.Progress
+			}
+			if cfg.Verbose != 0 {
+				opts.verbosity = cfg.Verbose
+			}
+
```

Comment:
I think we want flags to take precedence over the config file, since they're specified per-invocation.

Reply:
refactored